### PR TITLE
feat(auth): switch to OIDC authentication

### PR DESCRIPTION
**Pull request by open-dynaMIX**

### CODE REVIEW

**Comment by open-dynaMIX** [comment]
@winged I've decided against pulling in the whole keycloak and caddy stuff. It's a django cookiecutter after all ;)

**Review by winged — APPROVED** [review]
Awesome work! I do have some small suggestions, but feel free to ignore :)

File: {{cookiecutter.project_name}}/{{cookiecutter.project_name}}/models.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class UUIDModel(models.Model):
+    """
+    Models which use uuid as primary key.
+
+    Defined as {{cookiecutter.project_name}} default
+    """
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
```
[the remark this answers is not in the document]
Comment (open-dynaMIX) [reply {{cookiecutter.project_name}}/{{cookiecutter.project_name}}/models.py:13]: Noice! I've created an issue for this: #393

File: {{cookiecutter.project_name}}/{{cookiecutter.project_name}}/{{cookiecutter.django_app}}/models.py [hunk 2]
```diff
@@ -1,7 +1,99 @@
...
+
+
+class UserProfile(UUIDModel):
+    idp_id = models.CharField(max_length=255, unique=True, null=True, blank=False)
+    email = models.EmailField(unique=True, null=True, blank=True)
+    first_name = models.CharField(max_length=255, null=True, blank=True)
+    last_name = models.CharField(max_length=255, null=True, blank=True)
+
+    class Meta:
+        ordering = ("last_name", "first_name", "email")
```
[the remark this answers is not in the document]
Comment (open-dynaMIX) [reply {{cookiecutter.project_name}}/{{cookiecutter.project_name}}/{{cookiecutter.django_app}}/models.py:21]: True, done :+1:

**Pull request merged**